ee/api/vercel/vercel_proxy.py

```diff
@@ -0,0 +1,197 @@
+from rest_framework.permissions import AllowAny
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from posthog.exceptions_capture import capture_exception
+from posthog.models.organization_integration import OrganizationIntegration
+
+from ee.api.authentication import BillingServiceAuthentication, BillingServiceUser
+from ee.api.vercel.billing_proxy_region_mixin import BillingProxyRegionMixin
+
+logger = structlog.get_logger(__name__)
+
+ALLOWED_VERCEL_PATHS = frozenset(
+    [
+        "/billing/invoices",
+        "/billing/usage",
+    ]
+)
+
+VERCEL_API_BASE_URL = "https://api.vercel.com"
+REQUEST_TIMEOUT_SECONDS = 30
+
+
+class VercelProxyRequestSerializer(serializers.Serializer):
+    path = serializers.CharField(help_text="The Vercel API path to call (e.g., '/billing/invoices')")
+    method = serializers.ChoiceField(
+        choices=["GET", "POST", "PUT", "PATCH", "DELETE"],
+        help_text="HTTP method to use",
+    )
+    body = serializers.JSONField(required=False, default=dict, help_text="Request body to send to Vercel")
+
+    def validate_path(self, value: str) -> str:
+        normalized = urllib.parse.unquote(value)
+
+        if not normalized.startswith("/"):
+            raise serializers.ValidationError("Path must start with '/'")
+
+        if ".." in normalized:
+            raise serializers.ValidationError("Path traversal not allowed")
+
+        if normalized not in ALLOWED_VERCEL_PATHS:
+            raise serializers.ValidationError(
+                f"Path '{normalized}' is not in the allowlist of permitted Vercel API paths"
+            )
+
+        return normalized
+
+
+def _extract_access_token(integration: OrganizationIntegration) -> str:
+    token = integration.config.get("credentials", {}).get("access_token")
+    if not token:
+        raise ValueError(f"No access token found for integration {integration.integration_id}")
+    return token
+
+
+def forward_to_vercel(config_id: str, access_token: str, path: str, method: str, body: dict) -> requests.Response:
+    url = f"{VERCEL_API_BASE_URL}/v1/installations/{config_id}{path}"
+
+    headers = {
+        "Authorization": f"Bearer {access_token}",
+        "Content-Type": "application/json",
+    }
+
+    return requests.request(
+        method=method,
+        url=url,
+        headers=headers,
+        json=body if body else None,
+        timeout=REQUEST_TIMEOUT_SECONDS,
+    )
+
+
+class VercelProxyViewSet(BillingProxyRegionMixin, viewsets.ViewSet):
+    """
+    Proxy endpoint for the billing service to call Vercel APIs.
+
+    The billing service sends requests here with the Vercel API path and body.
+    PostHog validates the JWT, looks up the Vercel token from OrganizationIntegration,
+    and forwards the request to Vercel.
+
+    Region routing is handled by BillingProxyRegionMixin - if the integration
+    isn't found in US, the request is automatically proxied to EU.
```

Comment:
Not sure if there's a good reason for this but this adds more nodes and latency - we should call from billing to US or EU - we should be able to determine this based on `billing_customer.license_id` where `1` is EU and `2` is US.